internal/utils/miscellaneous.go

```diff
@@ -6,35 +6,32 @@ package utils
 
 import (
 	"fmt"
-	"time"
+	"maps"
+	"strings"
 
-	"github.com/robfig/cron/v3"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // MergeStringMaps merges the content of the newMaps with the oldMap. If a key already exists then
 // it gets overwritten by the last value with the same key.
 func MergeStringMaps(oldMap map[string]string, newMaps ...map[string]string) map[string]string {
-	var out map[string]string
+	allMaps := []map[string]string{oldMap}
+	allMaps = append(allMaps, newMaps...)
+	return MergeMaps[string, string](allMaps...)
+}
 
-	if oldMap != nil {
-		out = make(map[string]string)
-	}
-	for k, v := range oldMap {
-		out[k] = v
+// MergeMaps merges the contents of maps. All maps will be processed in the order
+// in which they are sent. For overlapping keys across source maps, value in the merged map
+// for this key will be from the last occurrence of the key-value.
+func MergeMaps[K comparable, V any](sourceMaps ...map[K]V) map[K]V {
```

Comment:
Same opinion expressed about generic function syntax being followed currently by callers of this function, that is expressed in https://github.com/gardener/etcd-druid/pull/777#discussion_r1594210522 applies here as well.

Reply:
Addressed, thanks.